File: python/bubble_sort.py
def bubble_sort(sequence):
    swaps = 0
    iterations = 0
    iter_swaps = 0
    previous, current = 0, 1
    while True:
        if sequence[previous] > sequence[current]:
            sequence[previous], sequence[current] = sequence[current], sequence[previous]
            swaps += 1
            iter_swaps += 1
        previous += 1
        current += 1
        if previous == len(sequence) - 1:
            iterations += 1
            if iter_swaps == 0:
                break
            iter_swaps = 0
            previous, current = 0, 1
    print(swaps)
    print(iterations)
    return sequence, swaps, iterations

File: js/bubbleSort.js
exports.bubbleSort = function (sequence) {
  let previous_swaps = 0;
  let iter = 0;
  let sorted = false;
  let swaps = 0
  while (sorted == false) {
    for (let i=0; i<sequence.length; i++) {
      if (i<sequence.length-1 && sequence[i+1] < sequence[i]) {
        let elem = sequence[i];
        sequence[i] = sequence[i+1];
        sequence[i+1] = elem;
        swaps += 1;
      }
    }
    if (swaps == previous_swaps) {
      sorted = true;
    }
    previous_swaps = swaps;
    iter += 1
  }
  return sequence
}